src/librustc/traits/error_reporting/on_unimplemented.rs

```diff
@@ -67,6 +67,24 @@ impl<'a, 'tcx> InferCtxt<'a, 'tcx> {
                     "a function"
                 })
             })
+        } else if let hir::Node::TraitItem(hir::TraitItem {
+            kind: hir::TraitItemKind::Method(_, hir::TraitMethod::Provided(body_id)),
+            ..
+        }) = &node
```

Comment:
why is this not a `match`?

Reply:
I just added to the existing branches to avoid touching more lines than necessary.

It certainly could be rewritten as a match.

Reply:
Seems like a good cleanup to make now as another commit.

Reply:
Already on it ;)